hw6/main.cpp

```diff
@@ -0,0 +1,151 @@
+#include <iostream>
+
+int main() {
+    // task 1
+    int first, step, last;
+    std::cout << "Enter first element of arithmetic progression: ";
+    std::cin >> first;
+    std::cout << "Enter step of arithmetic progression: ";
+    std::cin >> step;
+    std::cout << "Enter number of last element of arithmetic progression: ";
+    std::cin >> last;
+
+    if (first < 1) {
+        std::cout << "First element of arithmetic progression must be more or equal 1" << std::endl;
+    } else if (step < 1) {
+        std::cout << "Step of arithmetic progression must be more or equal 1" << std::endl;
+    } else if (last < first) {
+        std::cout << "Number of last element of arithmetic progression must be more than first element" << std::endl;
+    } else {
```

Comment:
wrong conditions. these parameters are valid:
-10, -10, 5
so you should print:
-10, -20, -30, -40, -50

only one valid check is the last input > 0